python/ray/data/_internal/logical/optimizers.py

```diff
@@ -31,13 +31,19 @@
 
 
 @DeveloperAPI
-def register_logical_rule(cls: Type[Rule]):
-    _LOGICAL_RULES.append(cls)
+def register_logical_rule(cls: Type[Rule], insert_index: Optional[int] = None):
+    if not insert_index:
```

Comment:
Should this be `if insert_index is not None`? If `insert_index == 0`, this'd append the rule rather than making it the first item, right?